neptunecontrib/monitoring/keras.py

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright (c) 2019, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+
+import neptune
+from neptune.exceptions import LibraryNotInstalled, NeptuneException
+
+try:
+    from keras.callbacks import Callback
+except ImportError:
+    try:
+        from tensorflow.keras.callbacks import Callback
+    except ImportError:
+        raise LibraryNotInstalled('Keras')
+
+
+class NeptuneMonitor(Callback):
+    """Logs Keras metrics to Neptune.
+
+    Goes over the `last_metrics` and `smooth_loss` after each batch and epoch
+    and logs them to appropriate Neptune channels.
+
+    See the example experiment here TODO
+
+    Args:
+        experiment: `neptune.Experiment`, optional:
+            Neptune experiment. If not provided, falls back on the current
+            experiment.
+        prefix: str, optional:
+            Prefix that should be added before the `metric_name`
+            and `valid_name` before logging to the appropriate channel.
+            Defaul is 'keras_'.
+
+    Examples:
+        Prepare data::
+
+            TODO update for keras
+
+        Now, create Neptune experiment, instantiate the monitor and pass
+        it to callbacks::
+
+            TODO update for keras
+
+    Note:
+        You need to have Keras or Tensorflow 2 installed on your computer to use this module.
+    """
+
+    def __init__(self, experiment=None, prefix='keras_'):
+        super().__init__()
+        self._exp = experiment if experiment else neptune
+        self._prefix = prefix
+
+    # Workaround for a Tensorflow-Keras incommpatibility issue https://github.com/keras-team/keras/issues/14125: 
+    def _implements_train_batch_hooks(self): return True
+    def _implements_test_batch_hooks(self): return True
+    def _implements_predict_batch_hooks(self): return True
+
+    def _send_metrics(self, logs, trigger):
+        if not logs:
+            return
+
+        prefix = self._prefix + trigger + '_'
+
+        for metric, value in logs.items():
+            try:
+                if metric in ('batch', 'size'):
+                    continue
+                name = prefix + metric
```

Comment:
As default `prefix="keras"` this suggests that I will have channels `keras_accuracy` or `keras_loss` which I don't like that much. 
Especially if I were to train some models with various libs (comparing pytorch with keras implementations for example) I may have to rename my logs. 

Is there any particular reason to have the default prefix set to `keras_` ?

Reply:
In fact, the way I implemented it, we'll have `keras_batch_accuracy`, `keras_batch_loss`, `keras_epoch_accuracy`, `keras_epoch_loss`, etc.

I have no particular preferences here, I roughly followed the naming in `neptune-tensorboard`, which uses `keras_on_batch_end_accuracy`, `keras_on_batch_end_loss` etc.

So you suggest we get rid of the keras prefix altogether?

Reply:
I got rid of the keras_ prefix, let me know if this works now.

Reply:
I'd drop that prefix unless it's a pain for some other reasons.